src/psyclone/tests/psyGen_test.py

```diff
@@ -569,6 +569,19 @@ def test_sched_view(capsys):
     assert colored("Schedule", SCHEDULE_COLOUR_MAP["Schedule"]) in output
 
 
+def test_sched_getitem():
+    '''Test that Schedule have the [int] operators overloaded to return the
+    given index child'''
+    _, invoke_info = parse(os.path.join(BASE_PATH,
+                                        "15.9.1_X_innerproduct_Y_builtin.f90"),
+                           api="dynamo0.3")
+    psy = PSyFactory("dynamo0.3", distributed_memory=True).create(invoke_info)
+
+    sched = psy.invokes.invoke_list[0].schedule
+    for indx in range(len(sched._children)):
+        assert sched[indx] is sched._children[indx]
```

Comment:
What happens with an out-of-bounds request? Would be good to test that it does what you expect.

Reply:
Added out-of-bounds test. Also range-indexing (`sched[:]`) test as it is also an option.

Reply:
Great.